modules/sdk/gradle-plugins-workspace/src/main/java/com/liferay/gradle/plugins/workspace/WorkspaceExtension.java

```diff
@@ -365,6 +375,53 @@ private String _decodeBundleUrl(ProductInfo productInfo) {
 		}
 	}
 
+	private String _getBundleChecksumMD5() {
+		String bundleUrl = getBundleUrl();
+
+		if (Objects.isNull(bundleUrl) || bundleUrl.isEmpty()) {
+			return null;
+		}
+
+		try {
+			DownloadCommand downloadCommand = new DownloadCommand();
+
+			downloadCommand.setCacheDir(_workspaceCacheDir);
+			downloadCommand.setPassword(null);
+			downloadCommand.setToken(false);
+			downloadCommand.setUrl(new URL(bundleUrl + ".MD5"));
+			downloadCommand.setUserName(null);
+			downloadCommand.setQuiet(true);
+
+			downloadCommand.execute();
+
+			Path md5Path = downloadCommand.getDownloadPath();
+
+			String md5Content = FileUtil.read(md5Path.toFile());
+
+			int lastSlash = bundleUrl.lastIndexOf('/');
+
+			if (lastSlash == -1) {
+				return null;
+			}
+
+			if (Objects.isNull(md5Content)) {
+				return null;
+			}
+
+			int md5Length = md5Content.indexOf(' ');
```

Comment:
What about md5 content that only contains the hash?  This will fail right and not be set?
like this MD5 file: https://releases.liferay.com/portal/7.2.0-m2/liferay-ce-portal-tomcat-7.2.0-m2-201901311950.tar.gz.MD5

So we need to support both simple MD5 content and md5 content that has spaces and filenames.